src/app/tabs/camera.cpp

```diff
@@ -0,0 +1,143 @@
+#include <app/config.hpp>
+#include <app/tabs/camera.hpp>
+#include <app/window.hpp>
+#include <app/theme.hpp>
+
+CameraTab::CameraTab(QWidget *parent) : QWidget(parent)
+{
+    QStackedLayout *stack = new QStackedLayout(this);
+    stack->addWidget(connect_widget());
+    stack->addWidget(cam_widget());
+
+    connect(this, &CameraTab::stream_connected, this, [stack]() {stack->setCurrentIndex(1);} );
+    connect(this, &CameraTab::stream_disconnected, this, [stack]() {stack->setCurrentIndex(0);} );
+
+    player = NULL;
+    // connect_stream(); TODO: offer auto-connect option
+}
+
+QWidget* CameraTab::cam_widget()

```

Comment:
pointer right aligned

Reply:
Yeah, I tried to get all of these. As a side point, the type is `QWidget*` and even though a lot of people do it, putting the * next to the variable is just weird (IMO, of course).

Reply:
probably just my C style showing through, but i see it more of a "this variable is a pointer of type" :p

Reply:
We're obviously in the realm of philosophy at this point but I will point out that both the C standard and the C++ standard refer to pointers as a type.